haystack/components/generators/chat/hugging_face_api.py

```diff
@@ -0,0 +1,237 @@
+        stop_words: Optional[List[str]] = None,
+        streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
+    ):
+        """
+        Initialize the HuggingFaceAPIChatGenerator instance.
+
+        :param api_type:
+            The type of Hugging Face API to use.
+        :param api_params:
+            A dictionary containing the following keys:
+            - `model`: model ID on the Hugging Face Hub. Required when `api_type` is `SERVERLESS_INFERENCE_API`.
+            - `url`: URL of the inference endpoint. Required when `api_type` is `INFERENCE_ENDPOINTS` or `TEXT_GENERATION_INFERENCE`.
+        :param token: The HuggingFace token to use as HTTP bearer authorization
+            You can find your HF token in your [account settings](https://huggingface.co/settings/tokens)
+        :param generation_kwargs:
+            A dictionary containing keyword arguments to customize text generation.
+                Some examples: `max_tokens`, `temperature`, `top_p`...
+                See Hugging Face's documentation for more information at: [chat_completion](https://huggingface.co/docs/huggingface_hub/package_reference/inference_client#huggingface_hub.InferenceClient.chat_completion).
+        :param stop_words: An optional list of strings representing the stop words.
+        :param streaming_callback: An optional callable for handling streaming responses.
+        """
+
+        huggingface_hub_import.check()
+
+        if isinstance(api_type, str):
+            api_type = HFGenerationAPIType.from_str(api_type)
+
+        api_params = api_params or {}
+
+        if api_type == HFGenerationAPIType.SERVERLESS_INFERENCE_API:
+            model = api_params.get("model")
+            if model is None:
+                raise ValueError(
+                    "To use the Serverless Inference API, you need to specify the `model` parameter in `api_params`."
+                )
+            check_valid_model(model, HFModelType.GENERATION, token)
+            model_or_url = model
+        elif api_type in [HFGenerationAPIType.INFERENCE_ENDPOINTS, HFGenerationAPIType.TEXT_GENERATION_INFERENCE]:
+            url = api_params.get("url")
+            if url is None:
+                raise ValueError(
+                    "To use Text Generation Inference or Inference Endpoints, you need to specify the `url` parameter in `api_params`."
+                )
+            if not is_valid_http_url(url):
+                raise ValueError(f"Invalid URL: {url}")
+            model_or_url = url
+        else:
+            raise ValueError(
+                f"Unsupported API type: {api_type}. Supported types are: {[e.value for e in HFGenerationAPIType]}"
+            )
+
+        # handle generation kwargs setup
+        generation_kwargs = generation_kwargs.copy() if generation_kwargs else {}
+        generation_kwargs["stop"] = generation_kwargs.get("stop", [])
+        generation_kwargs["stop"].extend(stop_words or [])
```

Comment:
Just for my own sake :-) stop is now unique across all HF APIs? Any need to update local chat generator?

Reply:
unfortunately not :smile:
`stop` only works for `chat_completion`